neps/search_spaces/yaml_search_space_utils.py

```diff
@@ -1,10 +1,13 @@
 from __future__ import annotations
 
+import logging
 import re
 
+logger = logging.getLogger(__name__)
+
 
 def convert_scientific_notation(value: str | int | float, show_usage_flag=False) \
-                                -> float | (float, bool):
+        -> float | (float, bool):
```

Comment:
First, use `tuple[float, bool]`, doing `(float, bool)` isn't valid syntax.

Also try to avoid `\` in code. Normally formatters and linters will fix this up in the following way which I believe is more readable for longer function signatures.

```python
def convert_scientific_notation(
    value: str | int | float,
    show_usage_flag: bool = False
) -> float | tuple[float, bool]: